### ISSUE TITLE
Inconsistent warnings and exceptions by _FeatureConsumer in Bio.GenBank

### ISSUE THREAD

**Issue by mdehoon** [biopython/biopython#4130]
### Setup

I am reporting a problem with Biopython version, Python version, and operating
system as follows:

```python
>>> import sys; print(sys.version)
3.9.10 (v3.9.10:f2f3f53782, Jan 13 2022, 16:55:46) 
[Clang 13.0.0 (clang-1300.0.29.30)]
>>> import platform; print(platform.python_implementation()); print(platform.platform())
CPython
macOS-10.15.7-x86_64-i386-64bit
>>> import Bio; print(Bio.__version__)
1.80.dev0
```

The `_FeatureConsumer` in `Bio.GenBank` handles different problems in the location string inconsistently:

```python
>>> from Bio.GenBank import _FeatureConsumer
>>> from Bio.SeqFeature import SeqFeature

>>> consumer = _FeatureConsumer(False)
>>> feature = SeqFeature()
>>> consumer._cur_feature = feature
```
1. If start is greater than end, a warning is printed and location is set to `None`:

```python
>>> consumer.location("90..10")
/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/biopython-1.80.dev0-py3.9-macosx-10.9-universal2.egg/Bio/GenBank/__init__.py:352: BiopythonParserWarning: It appears that '90..10' is a feature that spans the origin, but the sequence topology is undefined. Skipping feature.
  warnings.warn(
>>> print(feature.location)
None
```

2. A between location with a gap of more than one raises a `ValueError`:

```python
>>> consumer.location("10^20")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/biopython-1.80.dev0-py3.9-macosx-10.9-universal2.egg/Bio/GenBank/__init__.py", line 1198, in location
    cur_feature.location = _loc(
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/biopython-1.80.dev0-py3.9-macosx-10.9-universal2.egg/Bio/GenBank/__init__.py", line 340, in _loc
    raise ValueError(f"Invalid between location {loc_str!r}") from None
ValueError: Invalid between location '10^20'
```

3. A string containing both "order" and "join" raises a `LocationParserError`:

```python
>>> consumer.location("join(order(90..100,200..210))")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/biopython-1.80.dev0-py3.9-macosx-10.9-universal2.egg/Bio/GenBank/__init__.py", line 1264, in location
    raise LocationParserError(msg)
Bio.GenBank.LocationParserError: Combinations of "join" and "order" within the same location (nested operators) are illegal:
join(order(90..100,200..210))
```

All three are cases where the location string is incorrect. Depending on whether or not we want to skip features with an incorrect location string, we should either issue a warning, or raise an exception, in all three cases. Probably the cleanest is to raise a `LocationParserError` in all three cases, and let `location` ignore such errors by default.

**Comment by mdehoon** [comment]
@peterjc  I think this is easy to fix, we just need to decide what we want to do here.

**Comment by peterjc** [comment]
The ``ValueError`` should certainly be ``LocationParserError`` for consistency.

I too lean to an exception by default, but we've had long discussion about common corner cases where it was more pragmatic to give a warning and set the location to ``None``. This is often enough for the user to extract whatever the want from the file.

**Comment by mdehoon** [comment]
> I too lean to an exception by default, but we've had long discussion about common corner cases where it was more pragmatic to give a warning and set the location to None. This is often enough for the user to extract whatever the want from the file.

Btw note that the warning message is incorrect. It says "Skipping feature." but actually the feature is kept; just its location is set to `None`.

**Comment by peterjc** [comment]
Fair point, something like "skipping location" would be more accurate.

**Comment by mdehoon** [comment]
Fixed as part of #4128
